File: baekjoon/cpp/15655.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int N, M;
int arr[8]{};
int seq[8]{};

void dfs(int level) {
    
    if(level == M) {
        for(int i = 0; i < M; i++) cout << seq[i] << " ";
        cout << "\n";
        return;
    }
    
    for(int i = 0; i < N; i++) {
        if(level != 0 && arr[i] <= seq[level-1]) continue;
        seq[level] = arr[i];
        dfs(level + 1);
    }
}

int main()
{
    cin >> N >> M;
    
    for(int i = 0; i < N; i++){
        cin >> arr[i];
    }
    
    sort(arr, arr + N);
    
    dfs(0);
    return 0;
}

File: baekjoon/15655.py

n, m = map(int, input().split())
nums = sorted(list(map(int, input().split())))

arr = [0] * m

def dfs(st, level):
    if level == m : 
        print(*arr)
        return 
    
    for i in range(st, n):
        arr[level] = nums[i]
        dfs(i+1, level+1)

dfs(0, 0)

